src/openms/include/OpenMS/CHEMISTRY/SvmTheoreticalSpectrumGeneratorSet.h

```diff
@@ -61,28 +61,13 @@ namespace OpenMS
   class OPENMS_DLLAPI SvmTheoreticalSpectrumGeneratorSet
   {
 public:
-
-    /** @name Constructors and Destructors
-     */
-    //@{
-    /// Default constructor
-    SvmTheoreticalSpectrumGeneratorSet();
-
-    /// Copy constructor
-    SvmTheoreticalSpectrumGeneratorSet(const SvmTheoreticalSpectrumGeneratorSet & source);
-
-    /// Destructor
-    virtual ~SvmTheoreticalSpectrumGeneratorSet();
-    //@}
-
-    /// Assignment operator
-    SvmTheoreticalSpectrumGeneratorSet & operator=(const SvmTheoreticalSpectrumGeneratorSet & tsg);
+    // Rule of Zero in effect for this class.
```

Comment:
My preference would actually be 

```
    
     /// Default constructor
    SvmTheoreticalSpectrumGeneratorSet() = default;
    /// Copy constructor
    SvmTheoreticalSpectrumGeneratorSet(const SvmTheoreticalSpectrumGeneratorSet&) = default;               
    /// Move constructor
    SvmTheoreticalSpectrumGeneratorSet(SvmTheoreticalSpectrumGeneratorSet&&) = default;                    
    /// Copy assignment operator
    SvmTheoreticalSpectrumGeneratorSet& operator=(const SvmTheoreticalSpectrumGeneratorSet&) & = default;  
    /// Move assignment operator
    SvmTheoreticalSpectrumGeneratorSet& operator=(SvmTheoreticalSpectrumGeneratorSet&&) & = default;       
    /// Destructor
    virtual ~SvmTheoreticalSpectrumGeneratorSet() = default;    
```

also note that your change has the subtle effect of changing `SvmTheoreticalSpectrumGeneratorSet` from a class with a virtual destructor to one without one! This changes its size as well as the behavior of ptrs to derived classes. 

@timosachsenberg @jpfeuffer any comments regarding the rule of zero / rule of zero defaults / rule of five?

My preference is discussed here https://github.com/OpenMS/OpenMS/issues/3310#issuecomment-411141372 and see also http://scottmeyers.blogspot.com/2014/03/a-concern-about-rule-of-zero.html

Reply:
> Maybe a compromise would suit, a "Rule of All or Nothing": As long as you can, stick to the Rule of Zero, but if you have to write at least one of the Big Five, default the rest.

I can agree to that, I think it makes sense. Maybe the proper way to for this case would actually be to write a test to ensure that the move operators are present and use the "Rule of All or Nothing" so in that case we have clean code and if somebody decides to accidentally remove the move operator, it would trigger a test failure. Seems like the best of both worlds.

>  This PR is testing to see if this type of refactor would be accepted into the codebase in the future.

that sounds good and you caught us at a somewhat unprepared state where we have not come to complete conclusions in #3310 how to handle C++11 features. However, now would be a good time to conclude that discussion and your input is very welcome.

> `SvmTheoreticalSpectrumGeneratorSet` is a datatype (class) and not an interface[[1]](http://www.stroustrup.com/bs_faq2.html#virtual-dtor).

I agree in this case that removal of the virtual dtor is correct. It also makes class size larger and adds nothing. So I am happy to remove the virtual dtor, I just wanted to highlight how subtle differences can manifest in C++ when one is implicit instead of explicit ...